tensorstore/internal/http/http_request.h

```diff
@@ -88,19 +101,21 @@ class HttpRequestBuilder {
   /// response.
   HttpRequestBuilder& EnableAcceptEncoding();
 
+  /// Adds a `range` header to the http request if the byte_range
+  /// is specified.
+  HttpRequestBuilder& AddRangeHeader(OptionalByteRangeRequest byte_range, bool & result);
```

Comment:
To simplify encapsulation/inheritance, modified 

`bool AddXY(HttpRequestBuilder& request, args...)`

signatures to

`HttpRequestBuilder & HttpRequestBuilder(args..., bool result)`